### DISCUSSION TITLE
Proper usage of `LIKE` with UUID fields

### DISCUSSION THREAD

**Discussion by jeremyosterhoudt** [duckdb/duckdb#13781] [Q&A]
I'm running into an issue using `LIKE` with UUID fields.  Whenever a `LIKE` is used against a `UUID` in a filter/where clause I get the following exception:

```
Binder Error: No function matches the given name and argument types '~~(UUID, STRING_LITERAL)'. You might need to add explicit type casts.
	Candidate functions:
	~~(VARCHAR, VARCHAR) -> BOOLEAN
```

Here is an example to reproduce:
```sql
CREATE TABLE t1 (id INT, uid UUID)
INSERT INTO t1 VALUES (1, '7d2ebfdc-ab32-ed83-4e44-e8e43467fba4')
SELECT * FROM t1 WHERE uid like '%7d%'
```

I can cast the `UUID` field to a `VARCHAR` and this does work.  However, my question is this intended proper usage or a bug or a todo item, etc.?

**Comment by jgrg** [comment]
UUID fields are stored as `HUGEINT` (128 bit) integers, which is good for maximum storage efficiency and, I imagine, speed of equality comparisons.

Also nice for ensuring valid UUIDs. These all store `7d2ebfdc-ab32-ed83-4e44-e8e43467fba4`:
```sql
INSERT INTO t1 VALUES
    (2, '7d2ebfdcab32ed834e44e8e43467fba4')
  , (3, '7d2ebfdcab32ed834e44--------------e8e43467fba4')
  , (4, '7D2EBFDC-AB32-ED83-4E44-E8E43467FBA4');
```

If you do need to do substring lookups on UUIDs, you can store them as `VARCHAR` (in which case you'd probably want to strip out dash characters):
```sql
ALTER TABLE t1 ADD COLUMN uid_str VARCHAR;
UPDATE t1 SET uid_str = replace(uid::VARCHAR, '-', '');
```

I imagine the intention of UUID columns is that they're used for matching UUIDs, so string operations aren't intended to work on them.

**Comment by Mytherin** [comment]
This is intended - `UUID` columns are not text but numbers. The textual rendering is a hexadecimal variant of that number. You can perform string operations on them by explicitly converting them to varchar.
